zebrad/src/components/inbound/tests/real_peer_set.rs

```diff
@@ -782,3 +791,112 @@ async fn setup(
         listen_addr,
     )
 }
+
+#[cfg(feature = "getblocktemplate-rpcs")]
+mod submitblock_test {
+    use std::io;
+    use std::sync::{Arc, Mutex};
+    use tracing::Level;
+    use tracing_subscriber::fmt;
+
+    use super::*;
+
+    use crate::components::sync::PEER_GOSSIP_DELAY;
+
+    // Custom in-memory writer to capture logs
+    struct TestWriter(Arc<Mutex<Vec<u8>>>);
```

Comment:
Could you leave a note about the conditional compilation issue or how this is used?

Reply:
@oxarbitrage I believe this conversation needs to be resolved before this PR can merge